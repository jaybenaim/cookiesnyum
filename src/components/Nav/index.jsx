import React from "react";
import { logo } from "../../assets/images";
import "../../assets/stylesheets/nav.css";
import { Link, withRouter } from "react-router-dom";
import Auth from "../Auth";
import { filterGallery } from "../../redux/actions/galleryActions";
import { Icon, Navbar } from "react-materialize";
import { connect } from "react-redux";
import Search from "../Search";

const Nav = (props) => {
  const {
    user: { role },
  } = props;
  return (
    <Navbar
      alignLinks="right"
      brand={
        <Link className="brand-logo" to="/">
          <img
            src={logo}
            alt="logo cookie with a bite"
            height={140}
            width={140}
          />
        </Link>
      }
      centerChildren
      id="mobile-nav"
      menuIcon={<Icon>menu</Icon>}
      options={{
        draggable: true,
        edge: "left",
        inDuration: 250,

        outDuration: 200,
        preventScrolling: true,
      }}
    >
      <Search />

      <Link to="/" className="sidenav-close">
        {" "}
        Home{" "}
      </Link>
      <Link to="/products" className="sidenav-close">
        Products
      </Link>
      {role === "admin" && (
        <Link to="/admin" className="sidenav-close">
          Admin
        </Link>
      )}

      <Link to="/" className="sidenav-logo sidenav-close">
        {" "}
        <img
          src={logo}
          alt="logo cookie with a bite"
          height={140}
          width={140}
        />
      </Link>
      <Auth />
    </Navbar>
  );
};

const mapStateToProps = (state) => {
  return {
    user: state.auth.user,
  };
};

export default withRouter(connect(mapStateToProps, { filterGallery })(Nav));
