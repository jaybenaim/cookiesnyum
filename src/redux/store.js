import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import rootReducer from "./reducers";
import { createLogger } from "redux-logger";

const loggerMiddleware = createLogger();
const initialState = {};
const middleware = [thunk, loggerMiddleware];

const store = createStore(
  rootReducer,
  initialState,
  applyMiddleware(...middleware)
);
export default store;
