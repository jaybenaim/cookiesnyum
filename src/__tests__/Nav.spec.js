import React from "react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import Nav from "../components/Nav";
import { Provider } from "react-redux";
import store from "../redux/store";
import { HashRouter } from "react-router-dom";

describe("Nav", () => {
  it("should be primary color", () => {
    console.log = jest.fn();
    const tree = renderer
      .create(
        <Provider store={store}>
          <HashRouter basename="/dolcenada">
            <Nav />
          </HashRouter>
        </Provider>
      )
      .toJSON();
    const treeInstance = tree.root;

    expect(tree).toMatchSnapshot();
  });
});
