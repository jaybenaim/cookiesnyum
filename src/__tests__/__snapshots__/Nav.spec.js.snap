// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should be primary color 1`] = `
Array [
  <nav
    className=""
  >
    <div
      className="nav-wrapper container"
    >
      <a
        className="brand-logo brand-logo"
        href="#/dolcenada/"
        onClick={[Function]}
      >
        <img
          alt="logo cookie with a bite"
          height={140}
          src="logo.JPG"
          width={140}
        />
      </a>
      <a
        className="sidenav-trigger"
        data-target="mobile-nav"
        href="#!"
      >
        <i
          className="material-icons"
        >
          menu
        </i>
      </a>
      <ul
        className="hide-on-med-and-down right"
      >
        <li>
          <a
            className="sidenav-close"
            href="#/dolcenada/"
            onClick={[Function]}
          >
             
            Home
             
          </a>
        </li>
        <li>
          <a
            className="sidenav-close"
            href="#/dolcenada/products"
            onClick={[Function]}
          >
            Products
          </a>
        </li>
        <li>
          <a
            className="sidenav-logo sidenav-close"
            href="#/dolcenada/"
            onClick={[Function]}
          >
             
            <img
              alt="logo cookie with a bite"
              height={140}
              src="logo.JPG"
              width={140}
            />
          </a>
        </li>
        <li>
          <div
            className="auth-content"
          >
            <a
              className="btn btn-large  waves-effect waves-light hoverable auth-link sidenav-close"
              href="#/dolcenada/login"
              id="login"
              onClick={[Function]}
            >
              Log In
            </a>
            <a
              className="btn btn-large waves-effect waves-light hoverable accent-3 auth-link sidenav-close"
              href="#/dolcenada/register"
              id="register"
              onClick={[Function]}
            >
              Register
            </a>
          </div>
        </li>
      </ul>
    </div>
  </nav>,
  <ul
    className="sidenav right"
    id="mobile-nav"
  >
    <li>
      <a
        className="sidenav-close"
        href="#/dolcenada/"
        onClick={[Function]}
      >
         
        Home
         
      </a>
    </li>
    <li>
      <a
        className="sidenav-close"
        href="#/dolcenada/products"
        onClick={[Function]}
      >
        Products
      </a>
    </li>
    <li>
      <a
        className="sidenav-logo sidenav-close"
        href="#/dolcenada/"
        onClick={[Function]}
      >
         
        <img
          alt="logo cookie with a bite"
          height={140}
          src="logo.JPG"
          width={140}
        />
      </a>
    </li>
    <li>
      <div
        className="auth-content"
      >
        <a
          className="btn btn-large  waves-effect waves-light hoverable auth-link sidenav-close"
          href="#/dolcenada/login"
          id="login"
          onClick={[Function]}
        >
          Log In
        </a>
        <a
          className="btn btn-large waves-effect waves-light hoverable accent-3 auth-link sidenav-close"
          href="#/dolcenada/register"
          id="register"
          onClick={[Function]}
        >
          Register
        </a>
      </div>
    </li>
  </ul>,
]
`;
